tests/apollo/test_skvbc_view_change.py

```diff
@@ -394,6 +395,69 @@ async def test_single_vc_current_and_next_primaries_down(self, bft_network, trac
             num_consecutive_failing_primaries = 2
         )
 
+    @unittest.skipIf(environ.get('BUILD_COMM_TCP_TLS', "").lower() == "true", "Unstable on CI (TCP/TLS only)")
+    @with_trio
+    @with_bft_network(start_replica_cmd, rotate_keys=True)
+    async def test_replica_asks_to_leave_view(self, bft_network, exchange_keys=True):
+        """
+        This test makes sure that a single isolated replica will ask for a view change, but it doesn't occur:
+        1) Start all replicas
+        3) Isolate a single replica for long enough to trigger a ReplicaAsksToLeaveViewMsg
+        4) Let the replica rejoin the consensus
+        5) Make sure a view change does not happen and the isolated replica
+        rejoins the fast path in the existing view
+        """
+        
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+        n = bft_network.config.n
+
+        self.assertEqual(len(bft_network.procs), n,
+                         "Make sure all replicas are up initially.")
+
+        initial_primary = await bft_network.get_current_primary()
+
+        isolated_node = random.choice(
+            bft_network.all_replicas(without={initial_primary}))
+        isolated_replicas = set([isolated_node])
+
+        with net.ReplicaSubsetTwoWayIsolatingAdversary(bft_network, isolated_replicas) as adversary:
+            adversary.interfere()
+            try:
+                client = bft_network.random_client()
+                client.primary = None
+                for _ in range(5):
+                    msg = skvbc.write_req(
+                        [], [(skvbc.random_key(), skvbc.random_value())], 0)
+                    await client.write(msg)
+            except:
+                pass
+
+            # wait for 1.5 times the view change timeout to ensure the replica asks for a view change
+            await trio.sleep(int(viewChangeTimeoutMilli) / 666)
```

Comment:
When I combine the comment and the code, I get the equation `1,5 * x = x / 666`, which yields `x~31` (x, being the view change timeout). Something is fishy 😄

Reply:
10000/666=15 seconds

Reply:
isn't it easier to say `trio.sleep(seconds = 1.5 * viewChangeTimeoutMilli / 1000)` ?